src/fetchers/top-languages-fetcher.js

```diff
@@ -84,20 +86,34 @@ async function fetchTopLanguages(username, exclude_repo = []) {
 
       // if we already have the language in the accumulator
       // & the current language name is same as previous name
-      // add the size to the language size.
+      // add the size to the language size and increase repoCount.
       if (acc[prev.node.name] && prev.node.name === acc[prev.node.name].name) {
         langSize = prev.size + acc[prev.node.name].size;
+        repoCount = repoCount + 1;
+      }
+      else {
+        // reset repoCount to 1
+        // language must exist in atleast one repo to be detected
+        repoCount = 1;
       }
       return {
         ...acc,
         [prev.node.name]: {
           name: prev.node.name,
           color: prev.node.color,
           size: langSize,
+          count: repoCount,
         },
       };
     }, {});
 
+
+  Object.keys(repoNodes)
+    .forEach((name) => {
+      // comparison index calculation
+      repoNodes[name].size = Math.sqrt(repoNodes[name].size * repoNodes[name].count);
```

Comment:
Just a general note. The formula used here seems to be different from the formula you gave in https://github.com/anuraghazra/github-readme-stats/issues/1600#issuecomment-1046056305.

````
k = x^p * y^q
```

I think it still uses an old calculation.

```
x*y = k^2
k = sqrt(x*y)
```

In https://github.com/anuraghazra/github-readme-stats/issues/1600#issue-1143665672 you mention that this calculation is based on the h-index but other than the goal that you and this index have in mind I can not see similarities. If you are proposing this form in our case I don't think an `sqrt` is needed since the number of bytes that come out of Graphql will always be possitive.

Reply:
[The new method is inspired by the h-index](https://github.com/anuraghazra/github-readme-stats/issues/1600#issue-1143665672) - I thought of this while reading that [Wikipedia page]((https://en.wikipedia.org/wiki/H-index?wprov=sfla1)). I decided to err on the side of caution and reference that.

We get `k = sqrt(x*y)` when we use `p=0.5` and `q=0.5` in the general formula `k = x^p * y^q`.